### PR TITLE
feat: Improve project creation UI

### PR DESCRIPTION
**Pull request by romeonicholas**
AddModelComparison
ConfigureModelComparison
AddTeamComparison
ChooseSourceComparison
GitComparison
CreateNewProjectComparison

Photos show the old views on the left and updated views on the right. These changes also includes the removal of the unused "model initialization" step. Resolves issue #1110

### CODE REVIEW

File: frontend/src/styles.css [hunk 1]
```diff
@@ -176,27 +171,58 @@ fieldset > :not(.field-separator) {
...
+  width: 680px;
+}
+
+.same-line-input {
+  max-width: 185px;
+}
+.nextline-stepper .mat-step-label.mat-step-label {
+  text-align: center;
+  text-overflow: inherit;
+  white-space: normal;
```
Comment (MoritzWeber0) [inline frontend/src/styles.css:198]: We try to avoid global CSS classes where possible. Instead, use TailwindCSS.

The reasons are: 
- When code in the HTML templates are removed, CSS classes are often forgotten. 
- It's hard to keep an overview over available CSS classes. Everybody will define their own classes, which is against the concept of consistent and aligned classes.
- Changes to a global class can introduce several side effects.
- We don't have to invent classnames that nobody understands.

A page that is worth to read: https://tailwindcss.com/docs/utility-first

TailwindCSS has the nice concept of custom styles: https://tailwindcss.com/docs/adding-custom-styles, which can be used for the definition of a specific width. If it doesn't make sense to use TailwindCSS, prefer to use component scoped CSS classes.

**Review by MoritzWeber0 — CHANGES_REQUESTED** [review]

File: frontend/src/styles.css (hunk 1 above)
Comment (romeonicholas) [reply frontend/src/styles.css:198]: I'll move the lesser-used classes to the inline utility styles, but the form styling would probably be a mistake to split up--it's used across 9 forms in 9 different files which would ideally be templated/compontentized but would probably require more effort than it's worth. We can also talk about this at the next weekly, but honestly, I'm concerned the current approach is causing more problems than it's solving (that is, without a visual design spec you'll contniue to see more inconsistancies in styling with utility styles than with reusable classes).

Comment (romeonicholas) [reply frontend/src/styles.css]: OK--shifted everything inline except for the form class that gets used across so many files. I can move that inline if you prefer, but I think the class is more maintainable given the project creation flow will be updated/added to at some point and in the absense of a component the various forms are derived from it makes it easier to find them all. Or I can try to componentize everything, but that probably deserves a separate issue because it'd be structurally changing multiple parts of the site.

Comment (MoritzWeber0) [reply frontend/src/styles.css]: > We can also talk about this at the next weekly, but honestly, I'm concerned the current approach is causing more problems than it's solving (that is, without a visual design spec you'll contniue to see more inconsistancies in styling with utility styles than with reusable classes).

This might be more suited for discussion during the Collaboration Manager planning, but I can offer some insights based on our experience. We initially adopted both global and component-based styles, but this approach wasn't effective for us. The primary issue was the continuous addition of new classes, while the existing ones were seldom reused.

Style consistencies have never been a big priority and will continue to be less relevant than functionality and stability. In order to really have a consistent style, we need someone who has the time to actively enforce a consistent style. Regardless of whether we use inline or global styles. But we don't have the capacity for that, and I'm also worried that it could slow us down in the development process.

File: frontend/src/app/projects/create-project/create-project.component.html [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
 
-<div class="wrapper">
-  <mat-horizontal-stepper linear #stepper>
-    <mat-step completed="false" label="General information">
-      <div class="flex-center">
-        <mat-card class="section-card">
+<div class="wrapper flex justify-center">
+  <mat-horizontal-stepper class="w-[710px]" linear #stepper>
+    <mat-step completed="false" label="Create new project">
+      <div class="flex justify-center">
```
Comment (MoritzWeber0) [inline frontend/src/app/projects/create-project/create-project.component.html:9]: The background color seems to be slightly different between global background and the div: 
image

Let's have a quick chat about this after the Daily. I'd like to hear your opinion if we should use the same background color.

File: frontend/src/app/projects/models/create-model/create-model.component.html [hunk 3]
```diff
@@ -5,54 +5,55 @@
...
     linear
     #stepper
   >
-    <mat-step completed="false" label="Create model">
-      <app-create-model-base
-        [asStepper]="asStepper"
-        (create)="afterModelCreated()"
-      ></app-create-model-base>
-    </mat-step>
+    <div class="flex justify-center">
```
Comment (MoritzWeber0) [inline frontend/src/app/projects/models/create-model/create-model.component.html:13]: The component is not centered when triggered via the project details: 
image

image

File: frontend/src/app/projects/models/model-source/choose-source.component.html [hunk 4]
```diff
@@ -2,48 +2,65 @@
...
-        <app-mat-icon size="50px">link</app-mat-icon>
-      </mat-card>
-    </a>
-  </div>
+<div class="flex justify-center">
+  <mat-card class="mat-card-form" *ngIf="projectService.project$ | async">
+    <div
+      *ngIf="(projectService.project$ | async) && (modelService.model$ | async)"
+    >
+      <div
```
Comment (MoritzWeber0) [inline frontend/src/app/projects/models/model-source/choose-source.component.html:12]: Git sources should also be available for non-admins and Git sources don't require the T4C integration.

File: frontend/src/app/projects/models/model-source/choose-source.component.html [hunk 5]
```diff
@@ -2,48 +2,65 @@
...
-        <p>Skip this step</p>
-        <app-mat-icon size="50px">skip_next</app-mat-icon>
-      </mat-card>
-    </a>
-  </div>
+      <h3>Other</h3>
+      <mat-divider></mat-divider>
+      <div class="my-[10px]">
+        <a mat-button (click)="modelSourceSelection.emit('skip')">
+          <span> Manually configure model </span>
```
Comment (MoritzWeber0) [inline frontend/src/app/projects/models/model-source/choose-source.component.html:60]: `Manually configure model` isn't the right term for it. The button can be used to skip adding the model sources. Maybe something like: "Add the Model source later".

File: frontend/src/app/projects/create-project/create-project.component.html [hunk 6]
```diff
@@ -3,14 +3,14 @@
  ~ SPDX-License-Identifier: Apache-2.0
  -->
 
-<div class="wrapper">
-  <mat-horizontal-stepper linear #stepper>
-    <mat-step completed="false" label="General information">
-      <div class="flex-center">
-        <mat-card class="section-card">
+<div class="wrapper flex justify-center">
```
Comment (MoritzWeber0) [inline frontend/src/app/projects/create-project/create-project.component.html:6]: The mobile view has room for improvement. It's wasn't perfect on main either. To do so, you can add a few `max-w-mobile md:w-full` classes and add `mobile` to the `spacing` section of the `tailwind.config.js`. 
image

**Review by MoritzWeber0 — CHANGES_REQUESTED** [review]

File: frontend/src/app/projects/create-project/create-project.component.html (hunk 2 above)
Comment (romeonicholas) [reply frontend/src/app/projects/create-project/create-project.component.html:13]: Added shadow and rounded corners to stepper:
ShadedRoundedStepper

File: frontend/src/app/projects/models/model-source/choose-source.component.html (hunk 5 above)
Comment (romeonicholas) [reply frontend/src/app/projects/models/model-source/choose-source.component.html]: Updated:
ModelSourceSkipping

File: frontend/src/app/projects/create-project/create-project.component.html (hunk 6 above)
Comment (romeonicholas) [reply frontend/src/app/projects/create-project/create-project.component.html:6]: This one I'm kind of at a loss for. I don't think there's any way to continue using the horizontal stepper on mobile, so maybe I can flip it into a vertical stepper at a certain breakpoint? Removing the back buttons made sense on desktop but they're really important for mobile, but since we sometimes skip steps I don't know if that would require special logic, or do we leave them off?

File: frontend/src/app/projects/models/model-source/choose-source.component.html (hunk 4 above)
Comment (MoritzWeber0) [reply frontend/src/app/projects/models/model-source/choose-source.component.html]: Yes :)

File: frontend/src/app/projects/create-project/create-project.component.html (hunk 6 above)
Comment (romeonicholas) [reply frontend/src/app/projects/create-project/create-project.component.html:6]: Vertical also didn't look great, so I opted for horizontal with hidden labels and conditionally displayed headers so the user knew what step they were on. There's a weird media query for the stepper-lines and stepper headers, but apparenlty you can only target that with CSS, unless you know a workaround.
mobile

Comment (MoritzWeber0) [reply frontend/src/app/projects/create-project/create-project.component.html:6]: I really like the new mobile design, thanks for the implementation and the idea 👍

File: frontend/src/app/projects/models/create-model/create-model.component.html (hunk 3 above)
Comment (MoritzWeber0) [reply frontend/src/app/projects/models/create-model/create-model.component.html:14]: This change got lost with the last push, the model creation dialog has no shadow / rounded anymore: 

image


Also, the width of model description and model tool is not full anymore.

File: frontend/src/styles.css
Comment (MoritzWeber0) [inline frontend/src/styles.css:1]: This comment is not related to this file, I just need a file to open a conversation.

In the documentation, we have to update the "Create a Model" page and remove Step 4 (Choose Initialization).

File: frontend/src/app/projects/models/create-model/create-model.component.html (hunk 3 above)
Comment (romeonicholas) [reply frontend/src/app/projects/models/create-model/create-model.component.html:14]: Waah, replace-all error (the styling was actually still there but set to be sm instead of md), now fixed. Full-width applied to inputs again.

File: frontend/src/styles.css
Comment (romeonicholas) [reply frontend/src/styles.css:1]: Removed :white_check_mark:

**Review by MoritzWeber0 — APPROVED** [review]
Thanks for the great PR. A very nice UI improvement and nice improvements for mobile devices :)
